pyaedt/modeler/Object3d.py

```diff
@@ -4836,3 +4836,209 @@ def delete(self):
+    """
+
+    def __init__(self, primitives, name):
+        self._name = name
+        self._plane_coordinate_system = "Global"
+        self._color = None
+        self._root_point = None
+        self._normal = None
+        self._primitives = primitives
+        self._all_props = None
+
+    @property
+    def m_Editor(self):
+        """Pointer to the oEditor object in the AEDT API. This property is
+        intended primarily for use by FacePrimitive, EdgePrimitive, and
+        VertexPrimitive child objects.
+
+        Returns
+        -------
+        oEditor COM Object
+
+        """
+        return self._primitives.oeditor
+
+    @property
+    def logger(self):
+        """Logger."""
+        return self._primitives.logger
+
+    @property
+    def name(self):
+        """Name of the plane as a string value.
+
+        Returns
+        -------
+        str
+           Name of object as a string value.
+
+        References
+        ----------
+
+        >>> oEditor.GetPropertyValue
+        >>> oEditor.ChangeProperty
+
+        """
+        return self._name
+
+    @name.setter
+    def name(self, plane_name):
+        if plane_name not in self._primitives.planes.keys():
+            plane_old_name = self._name
+            if plane_name != self._name:
+                name_property = []
+                name_property.append("NAME:Name")
+                name_property.append("Value:=")
+                name_property.append(plane_name)
+                changed_property = ["NAME:ChangedProps", name_property]
+                property_servers = ["NAME:PropServers"]
+                property_servers.append(self._name)
+                plane_tab = ["NAME:Geometry3DPlaneTab", property_servers, changed_property]
+                all_tabs = ["NAME:AllTabs", plane_tab]
+                _retry_ntimes(10, self._primitives.oeditor.ChangeProperty, all_tabs)
+                self._name = plane_name
+                # TO BE DELETED self._primitives.cleanup_objects()
+                # Update the name of the plane in the ``planes`` dictionary listing all existing planes.
+                self._primitives.planes[plane_name] = self._primitives.planes.pop(plane_old_name)
+        else:
+            self.logger.warning("A plane named '%s' already exists.", plane_name)
+
+    @property
+    def valid_properties(self):
+        """Valid properties.
+
+        References
+        ----------
+
+        >>> oEditor.GetProperties
+        """
+        if not self._all_props:
+            self._all_props = _retry_ntimes(10, self.m_Editor.GetProperties, "Geometry3DPlaneTab", self._name)
+        return self._all_props
+
+    # Note: We currently cannot get the color property value because
+    # when we try to access it, we only get access to the 'edit' button.
+    # Following is the line that we would use but it currently returns 'edit'.
+    # color = _retry_ntimes(10, self.m_Editor.GetPropertyValue, "Geometry3DPlaneTab", self._name, "Color")
```

Comment:
```suggestion
    # Note: You currently cannot get the color property value because
    # when you try to access it, you only get access to the 'edit' button.
    # Following is the line that you would use, but it currently returns 'edit'.
    # color = _retry_ntimes(10, self.m_Editor.GetPropertyValue, "Geometry3DPlaneTab", self._name, "Color")
```